sqldelight-compiler/src/main/kotlin/com/squareup/sqldelight/SqliteCompiler.kt

```diff
@@ -46,6 +47,7 @@ class SqliteCompiler {
       projectPath: String
   ): Status {
     try {
+      nameAllocators.put(parseContext.start.inputStream, NameAllocator())
```

Comment:
Ah, I missed that.

The current behavior is basically creating global variables though which I think is worse than passing instances around. We could take the approach where we could change this `write` method to be static and have the `SqliteCompiler` create per call to `write` and host the `NameAllocator`.